### PR TITLE
Refine contributing guide

### PR DESCRIPTION
**Pull request by danielrbradley**
- Update tool versions.
- Add more information about the versioning process.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -54,48 +53,74 @@ Azure's API specifications are organised into "Resource Providers" (or "Services
...
 
 - `tracking` specifies a single version of the API to fetch all resources from.
 - `additions` specified a map of specific resource (or invoke) name to API version to include in the default version.
 
 Both `tracking` and `additions` can be specified together but if the set of resources overlaps (the same resource exists) in both the tracking version and the additions, then the versioning program will fail with an error.
 
-From the spec YAML file, we can generate a version 'lock' JSON file ([`v1-lock.json`](./versions/v1-lock.json) and [`v2-lock.json`](./versions/v2-lock.json)) which contain the fully expanded set of default resources. These are the files used at the point of generating the schema.
+From the Spec file, we generate the Lock JSON file (e.g. [`v2-lock.json`](./versions/v2-lock.json)) which contain the fully expanded set of default resources. These are the files used at the point of generating the schema.
+
+### Additional Files
```
Comment (thomas11) [inline CONTRIBUTING.md:99]: empty section?

Comment (danielrbradley) [reply CONTRIBUTING.md:99]: Ah left-over after moving the content into "Additional Reports"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -54,48 +53,74 @@ Azure's API specifications are organised into "Resource Providers" (or "Services
...
 
 ### Type Compatibility
 
 Supporting every version of every API causes the SDK size to be very large. One method we've used to reduce the required size is to only have a single set of types for each API which are re-used across all versions. However, not all the types are compatible and therefore a type with the same name might be incompatible in the next version. This incompatibility is currently detected but ignored on a case-by-case basis for now.
 
 Ideally, at the point a new incompatible version of the type is added, it should then be created with a unique, stable name. Alternatively, we could create a union of the two possible types
 
+### Default Version Locks
+
+The default version is calculated and written to a 'lock' file which list every resource (or invoke) at a specific API version. This file is read in during the schema and SDK generation. Currently there is a `v1-lock.json` and `v2-lock.json` lock file. These lock files should not be edited directly, but should be calculated by the versioner tool using a specific algorithm.
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply CONTRIBUTING.md:117]: Sorry, not sure what you mean

[the remark this answers is not in the document]
Comment (thomas11) [reply CONTRIBUTING.md:115]: >a 'lock' file which _lists_ every ...

nbd though

**Pull request merged**